custom/icds_reports/reports/disha.py

```diff
@@ -28,8 +29,18 @@ def __init__(self, state_name, month):
     def _blob_id(self):
         return 'disha_dump-{}-{}.json'.format(self.state_name.replace(" ", ""), self.month.strftime('%Y-%m-%d'))
 
-    def get_data(self):
-        dump = IcdsFile.objects.filter(blob_id=self._blob_id()).first()
+    @memoized
+    def _get_file_ref(self):
+        return IcdsFile.objects.filter(blob_id=self._blob_id()).first()
+
+    def export_exists(self):
```

Comment:
looks like this can be just `return self._get_file_ref()`
or you can also rename it to `state_data_file_for_month` and use a single method instead of two

Reply:
This method allows for cleaner interface, which is used [here](https://github.com/dimagi/commcare-hq/pull/22463/files#diff-48e45a26c4763ddeafc50cc7f7e97a12R112).

Reply:
Not blocking but i think `state_data_file_for_month` would work there

`if state_data_file_for_month and not rebuild:`